code/src/main/java/com/onlinecontacttracing/servlets/get-data/CalendarDataForNegativeUser.java

```diff
@@ -1,16 +1,87 @@
 package com.onlinecontacttracing.authentication;
 
 import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.DateTime;
+import com.google.api.services.calendar.Calendar;
+import com.google.api.services.calendar.model.Event;
+import com.google.api.services.calendar.model.Events;
+import com.onlinecontacttracing.storage.Constants;
+import com.onlinecontacttracing.storage.NegativeUserPlace;
+import java.util.List;
+import java.util.Collections;
+import java.util.Optional;
+import java.util.logging.Logger;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlacesApi;
+import com.google.maps.model.PlacesSearchResult;
+import com.googlecode.objectify.Objectify;
 
 class CalendarDataForNegativeUser implements Runnable {
+
+  private static final String APPLICATION_NAME = "Get Calendar Data From Negative User";
+  private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+  private static final String calendarType = "primary";
+  static final Logger log = Logger.getLogger(CalendarDataForNegativeUser.class.getName());
+
+  private final Objectify ofy;
+  private final String userId;
   private final Credential credential;
 
-  public CalendarDataForNegativeUser(Credential credential) {
+  public CalendarDataForNegativeUser(Objectify ofy, String userId, Credential credential) {
+    this.ofy = ofy;
+    this.userId = userId;
     this.credential = credential;
   }
 
   @Override
   public void run() {
-    // TODO Get contacts from people api
+    try {
+      final NetHttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+
+      // Get user's calendar
+      Calendar service = new Calendar.Builder(httpTransport, JSON_FACTORY, credential)
+        .setApplicationName(APPLICATION_NAME)
+        .build();
+
+      // Set up Places API
+      GeoApiContext context = new GeoApiContext.Builder()
+        .apiKey("AIzaSyBMrfBNGcVEtoRsoduXvYSjd9piD36W7Qg")
```

Comment:
I think this might want to be a private secret - do you know whether this is something we need to keep in a secret file (similar to our credentials?)

Reply:
Yeah, I will see if I can extract it from the json file.